src/microbit/model/display.py

```diff
@@ -58,13 +63,20 @@ def scroll(self, value, delay=150, wait=True, loop=False, monospace=False):
         while True:
             # Show the scrolled image one square at a time.
             for x in range(appended_image.width() - CONSTANTS.LED_WIDTH + 1):
+                self.__image_lock.acquire()
                 self.__image.blit(
                     appended_image, x, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT
                 )
+                self.__image_lock.release()
+
                 # If show or scroll is called again, there will be a different pid and break
+                self.__thread_identifier_lock.acquire()
                 if self.__current_pid != threading.get_ident():
                     break
+                self.__thread_identifier_lock.release()

```

Comment:
If the condition is satisfied and we break, is `self.__thread_identifier_lock.release()` called?

Reply:
Very true. Need to change that